plumbing/format/packfile/packfile.go

```diff
@@ -312,18 +344,24 @@ func (p *Packfile) fillREFDeltaObjectContent(obj plumbing.EncodedObject, ref plu
 	obj.SetType(base.Type())
 	err = ApplyDelta(obj, base, buf.Bytes())
 	p.cachePut(obj)
-	bufPool.Put(buf)
 
 	return err
 }
 
 func (p *Packfile) fillOFSDeltaObjectContent(obj plumbing.EncodedObject, offset int64) error {
 	buf := bytes.NewBuffer(nil)
```

Comment:
Shouldn't we get the buffer from the pool instead of creating a new one?

```go
buf := bufPool.Get().(*bytes.Buffer)
```

Reply:
Yes, I missed that one. I'll fix it in a minute.